### PR TITLE
Remove unnecessary conditional defines from eventkit.cs

### PR DESCRIPTION
**Pull request by chamons**
- Unlike the rest of my recent work, this is all done by hand, so please review closely.
- Part of https://github.com/xamarin/xamarin-macios/issues/14444

### CODE REVIEW

File: src/eventkit.cs [hunk 1]
```diff
@@ -319,11 +328,13 @@ interface EKEvent {
 		NSDate OccurrenceDate { get; }
 
 #if MONOMAC
```
Comment (rolfbjarne) [inline src/eventkit.cs:330]: Any reason to keep the `#if MONOMAC` here?

**Review by rolfbjarne — APPROVED** [review]

File: src/eventkit.cs (hunk 1 above)
Comment (chamons) [reply src/eventkit.cs:330]: I thought I did, but I don't see why now. Let me try removing it.

**Review by mandel-macaque — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

File: src/eventkit.cs (hunk 1 above)
Comment (chamons) [reply src/eventkit.cs]: Done!

**Comment by chamons** [comment]
Test failures unrelated https://github.com/xamarin/xamarin-macios/pull/14587

**Pull request merged**